src/OmniSharp.Script/ScriptProjectSystem.cs

```diff
@@ -137,29 +114,28 @@ private ProjectInfo CreateCsxProject(string csxPath, IScriptServicesBuilder base
                 return Context.CsxFileProjects[csxPath];
             }
 
-            // Process the file with ScriptCS first
             Logger.LogInformation($"Processing script {csxPath}...");
             Context.CsxFilesBeingProcessed.Add(csxPath);
-            var localScriptServices = baseBuilder.ScriptName(csxPath).Build();
-            var processResult = localScriptServices.FilePreProcessor.ProcessFile(csxPath);
+
+            var fileParser = new FileParser(Context.RootPath);
```

Comment:
yeah. The only "hurdle" is if I encounter a `#load foo.csx` directive, I need to follow it and pick up the `foo.csx` and that one might have it's own `#loads` and so on.

Scripting APIs would use [this](https://github.com/dotnet/roslyn/blob/master/src/Scripting/Core/ScriptSourceResolver.cs#L11) to resolve that whole dependency tree so maybe that would work, to parse the entry file and then follow it's `#loads` using `ScriptSourceResolver` recursively.